Openhab.Core/Model/OpenHABItem.cs

```diff
@@ -100,5 +139,24 @@ private void ParseNode(XElement startNode)
             State = startNode.Element("state")?.Value;
             Link = startNode.Element("link")?.Value;
         }
+
+        public void UpdateValue (object value)
+        {
+            if (value != null)
+            {
+                string newValue = value.ToString() + this.Unit;
+                Messenger.Default.Send(new TriggerCommandMessage(this, newValue));
+                _state = newValue;
+            }
+        }
+
+        public double GetStateAsDoubleValue()
+        {
+            string newstate = Regex.Replace(_state, "[^0-9,.]", string.Empty);
+            double value = 0;
+            _ = double.TryParse(newstate, out value);
```

Comment:
@hoffe86 What would you suggest to do if the parseing failed?
I initialized the "value" variable with 0.
If TryParse fails it will stay 0.

Or do you mean that i should display an error / log it if something goes wrong?

Reply:
@cd-tronic setting to 0 sound reasonable for me and I would log this issue as warning.
For retrieving a logger instance from the Dependency Injection you can use the following call without changing the OpenHABItem constructor:

`ILogger<OpenHABItem> logger = (ILogger<OpenHABItem>)DIService.Instance.Services.GetService(typeof(ILogger<OpenHABItem>));`

Reply:
hi @hoffe86,
i tried to add this but VS doesn't find OpenHAB.Windows Namespace if i'm working in OpenHAB.Core what i think is by design. 
Do you know any way to add a log message from Core Project?

Reply:
@cd-tronic you are right this only works in the UI project.
The correct way in this case would be to introduce a parameter of type ILogger<OpenHABItem> in the constructor and the Dependency Injection would initialize this parameter.

For the OpenHABItem class this will not work, because the parameterless constructor is used by the JSON deserializer and so we have no change to inject a ILogger via DI. From my side it is fine to keep it as it is without logging